pages/api/course/[id]/posts.ts

```diff
@@ -0,0 +1,35 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { HttpMethod } from "@lib/http-method";
+import { isValidUUID } from "@lib/helpers/backend/valid-uuid";
+import { getPostsForCourse } from "@lib/services/course";
+import { StatusCodes } from "http-status-codes";
+import { isAuthenticated } from '@lib/helpers/backend/user-helpers';
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  switch (req.method as HttpMethod) {
+    case HttpMethod.GET:
+      return getPostsForCourseHandler(req, res);
+    default:
+      return res.status(StatusCodes.METHOD_NOT_ALLOWED).end();
+  }
+}
+
+async function getPostsForCourseHandler(req: NextApiRequest, res: NextApiResponse) {
+  const { id, length, page } = req.query;
+  const courseID = id as string;
+  
+  if (!isValidUUID(courseID)) {
+    return res.status(StatusCodes.BAD_REQUEST).end();
+  }
+
+  const lengthInt = length ? parseInt(length as string) : 20;
```

Comment:
Maybe it makes sense to paginate the results in this method instead of the helper method?